HBHE M2-parameters: chi2, energy, pedSigma in M2

mariadalfonso opened the pull request:
This PR adjusts some of the M2 parameters.

1. Follow up on the siPM parameters changes happened in GT 81X_upgrade2017_realistic_v23 and PR#16526 (i.e. pedestal width and 20GeV threshold both passed in fC)

Also discussed the chi2 threshold for the 1 vs 3 pulses but not merged in this PR.
No effects expected in the run2 workflow

Validation plots in 
http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf

mariadalfonso committed:
changes: chi2, energy, pedSigma in M2

cmsbuild commented:
A new Pull Request was created by  @mariadalfonso  for CMSSW_8_1_X.

It involves the following packages:

RecoLocalCalo/HcalRecAlgos
RecoLocalCalo/HcalRecProducers


@cmsbuild, @cvuosalo, @slava77, @davidlange6 can you please review it and eventually sign? Thanks.
@argiro this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented on RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc:
```diff
@@ -246,12 +246,13 @@ void PulseShapeFitOOTPileupCorrection::setPUParams(bool   iPedestalConstraint, b
 						   double iNoiseHPD,double iNoiseSiPM,
 						   double iTMin,double iTMax,
-						   double its4Chi2,
+						   std::vector<double> its4Chi2,
```
pass by const reference, not by value

slava77 commented:
http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
p5 has fewer events at the high end.

Did you use the same number of events for testing or is it because there are points in the upper plot that are not displayed?

Given that chi2=5000 is a rather nonsensical value, can you come up with some energy-based balance to decide where to switch to 3 pulses? 
If we stay with this extremely large value of chi2: is there some strong evidence that this is representative of what we will see in data?

slava77 commented on RecoLocalCalo/HcalRecProducers/python/HBHEMethod2Parameters_cfi.py:
```diff
@@ -8,21 +8,21 @@
     applyTimeSlew         = cms.bool(True),           #units
     ts4Min                = cms.double(0.),           #fC
-    ts4Max                = cms.vdouble(100.,70000.), #fC # this is roughly 20 GeV
+    ts4Max                = cms.vdouble(100.,45000.), #fC # this is roughly 20 GeV
```
how stable is this channel to channel?

kpedro88 replied:
For the moment, all the channels have the same gain in MC conditions. At some point, it might be better to make this database-dependent, but it was pointed out that such a change would also impact M2 results for HB with HPDs (and HE pre-2017), since it hasn't been done that way so far.

slava77 replied:
what kind of variation can we expect here?
What was it for HPDs?
Is it o(10%) or O(100%)?

slava77 commented:
on p5 of the slides I see larger digressions from m2/m0~1. 
Is this from the increase in pedSigmaSiPM  or just random? Maybe a larger stat plot would be more descriptive.

Please provide the absolute scale plots (plots in #16526 were suggestive that it gets worse)

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 9f8dc47a7872eabdf9f9e7d9e7f57670f738556b

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16328/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.731 step2
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step2_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>10021.0 step2
<pre>runTheMatrix-results/10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step2_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10024.0 step2
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-13-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Mon Nov 14 12:31:08 2016-date Mon Nov 14 12:23:47 2016 s - exit: 18688
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Mon Nov 14 12:31:08 2016-date Mon Nov 14 12:23:47 2016 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-13-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Mon Nov 14 12:30:07 2016-date Mon Nov 14 12:24:12 2016 s - exit: 18688
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Mon Nov 14 12:30:07 2016-date Mon Nov 14 12:24:12 2016 s - exit: 18688

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

mariadalfonso commented:
Sorry, getting this to fix now

----- Begin Fatal Exception 14-Nov-2016 13:08:22 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=HcalHitReconstructor label='hltHbherecoMethod2L1EGSeeded'
Exception Message:
ValueError type of ts4chi2 is expected to be vector<double> but declared as double
----- End Fatal Exception -------------------------------------------------

slava77 commented:
On 11/14/16 4:18 AM, mariadalfonso wrote:
> Sorry, getting this to fix now
> 
> ----- Begin Fatal Exception 14-Nov-2016 13:08:22 CET-----------------------
> An exception of category 'Configuration' occurred while
> [0] Constructing the EventProcessor
> [1] Constructing module: class=HcalHitReconstructor
> label='hltHbherecoMethod2L1EGSeeded'
> Exception Message:
> ValueError type of ts4chi2 is expected to be vector but declared as double
> ----- End Fatal Exception -------------------------------------------------
> 

Is it normal for HLT to even depend on M2 parameters?


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-260321031>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbtM8UVJSg-rk3B0vu4IlgrFZDyLOks5q-FGFgaJpZM4KxFgT>.
>

kpedro88 committed:
customize HLT for changed M2 param

mariadalfonso committed:
change std::vector<double> into const std::vector<double> &

mariadalfonso commented:
@slava77

I updated the code to follow your suggestion.

Indeed the HLT now depend on the M2: this year the Egamma group updated the H/E cut to use locally the M2 instead the M3. There are a bunch of menu configuration in HLTrigger/Configuration,  those should be updated on the fly give change in the release but seems not. Still need to figure out how update these w/o manually edit each one.

I updated the slides:
1. the conditions in the GT23 seems worst based on the isolated track studies and overall M2-chi2
2. for studies in the updated slides, I recycled the same genStep and redo the step2 (DIGI) and step3 (RECO) since the GT23 update both.

slava77 commented:
@mariadalfonso 

For customization of HLT, I think you should use HLTrigger/Configuration/python/customizeHLTforCMSSW.py
@fwyzard @Martin-Grunewald please confirm

cmsbuild commented:
Pull request #16569 was updated. @cmsbuild, @cvuosalo, @slava77, @davidlange6 can you please check and sign again.

slava77 commented:
@mariadalfonso 
about the chi2=5000 switch:
please clarify what's plotted on p3 in red on the right. 
Is it one-pulse chi2 (with some manual setting to enforce fit of single pulse)?

If I understand the notation correctly: red on the left plot corresponds to blue on the right side. Is this correct?

How does the energy distribution look like with PU50?
If I so far understood the plots correctly, the discontinuity at 20 GeV should still be there because almost all hits will be fitted with 3 pulses given the red distribution.

kpedro88 commented:
The HLT configs are auto-generated through some mysterious process. I think you might need to add a customize function for HLT. I sent you https://github.com/mariadalfonso/cmssw/pull/1 with something that appears to work.

mariadalfonso committed:
Merge pull request #1 from kpedro88/M2-param-Fixes

customize HLT for changed M2 param

cmsbuild commented:
Pull request #16569 was updated. @perrotta, @cmsbuild, @silviodonato, @cvuosalo, @fwyzard, @Martin-Grunewald, @slava77, @davidlange6 can you please check and sign again.

kpedro88 commented:
@slava77 it's quite difficult to make any comparison to data. We only have the TB data from last year, but we know now that the collision data will not be the same (due to the change in overvoltage, which is the reason for the recent SiPM parameter updates).

kpedro88 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

mariadalfonso commented:
@slava77 
I update slide 3: central plot is with the ttbar w/ PU50, same feature of the step at 20GeV
http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
The ttbar plots on slide3 (both energies and the chi2), are done with the relval already re-recoed (so no manual setting).

slava77 commented:
On 11/14/16 10:48 AM, mariadalfonso wrote:
> @slava77 <https://github.com/slava77>
> I update slide 3: central plot is with the ttbar w/ PU50, same feature
> of the step at 20GeV
> http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
> The ttbar plots on slide3 (both energies and the chi2), are done with
> the relval already re-recoed (so no manual setting).

p7 is important, but is also unclear:
- why are the error bars so much larger on the right side for sigma/mean
plots?

The upper plot pair is a bit mixed: it looks like the response is above
1 now, but then the change in the size of the error bars between left
and right might be suggestive that the uncertainties are not assigned
correctly on the points.
(maybe the points are all stat consistent)
Could you please check.



> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-260424570>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbmxMAXNaDxau_y0r7S34_d-vWnixks5q-Kz8gaJpZM4KxFgT>.
>

cmsbuild commented:
+1
Tested at: 10340ecf7a9684f8a444baa38f21c1e2176fdbcd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16338/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/10340ecf7a9684f8a444baa38f21c1e2176fdbcd
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16338/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16338/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16338/summary.html

Martin-Grunewald commented:
+1

mariadalfonso commented:
@slava77
I added comparison the M0 energy vs M2 and GT22 vs GT23 on top of this PR
see slide 8 of 
http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
With the limited statistics: I do not see any mis behavior of the M2 vs M0 or the GT23 vs GT22.
Also not a significant improvement in the energy response neither.
Maybe some of your DQM plot will help more?
Is there something else needed from myside for the GT23 and this PR to go ?

The lost in statistics in the isotrack response is not due to the GT23 and PR changes.. 
I suspect that are due to the recent changes in the Calibration/HcalCalibAlgos Calibration/HcalCalibAlgos (@bsunanda)

@kpedro88

mariadalfonso committed:
Merged M2-param-Fixes from repository mariadalfonso

mariadalfonso commented:
@slava77 
I finally solved the statistics problem of the isolated particle (I just had to redo the step1(GEN) on top of the 8_1_0_pre16).

I updated slide 7/8 of http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
1) The absolute scale/resolution is little better than previous pre_releases in the range 30-50 GeV with this new GT/PR
2) the M2 scale is better than the M0 
Things looks good GT23 + thisPR.

@slava77  please let me know if there is something else from my side for the 8_1_0 
(Improving the chi2 and the CPU time for the end of the year is on top of the to do list, but is takes time)

@kpedro88

slava77 commented:
On 11/16/16 5:13 PM, mariadalfonso wrote:
> @slava77 <https://github.com/slava77>
> I finally solved the statistics problem of the isolated particle (I just
> had to redo the step1(GEN) on top of the 8_1_0_pre16).
> 
> I updated slide 7/8 of
> http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
> 1) The absolute scale/resolution is little better than previous
> pre_releases in the range 30-50 GeV with this new GT/PR
> 2) the M2 scale is better than the M0
> Things looks good GT23 + thisPR.
> 
> @slava77 <https://github.com/slava77> please let me know if there is
> something else from my side for the 8_1_0
> (Improving the chi2 and the CPU time for the end of the year is on top
> of the to do list, but is takes time)
> 
> @kpedro88 <https://github.com/kpedro88>

It's nice to see that the response is actually OK (without PU).

I'm still trying to digest p3.
Is the middle plot with the chi2=5000 switch or is it with the old
chi2=15 (or is it going to be about the same,
if the red distribution on the right is for the single pulse).

Maybe the RHS (chi2) plot in slices of ts45 would be more informative.

The fact that the chi2 is so poor means that once we switch to 3-fit
pulse we can loose a lot of energy.

I still think/hope that around ts45 of 15 to 20 GeV (or even above ~5-7 GeV)
 the HE chi2 looks more like blue rather than red.
If it's not the case and the component above chi2 of 5000 is still
large, we are making a mistake to have a switch here.


recall that the main purpose of M2 is to get the energy right in
presence of OOT PU.



> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261125051>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbjJvVJrw6Fal18kRVxhyeruAmxmAks5q-6okgaJpZM4KxFgT>.
>

mariadalfonso commented:
@slava77 
p3 show the picture out of the box from the 8_0_1_pre16 relVal for TTbar
p5 is show the energy spectrum before (in a non pU scenario) and after the change of chi2=15 and chi2=5000

With the GT23 the noise is reduced quite a bit, so any difference between the DIGI/template need to be compared with a lower noise: the "chi2" with GT23 is now lower than the GT22. 

I can try to check what happen when we have always the 3 pulse fit for all the energy range.

mariadalfonso commented:
@kpedro88 @abdoulline 
Do the current template 203 need to be updated to follow the analysis of the 2.8 mm and 3.3 mm adjustements you have here ?
https://indico.cern.ch/event/586254/contributions/2361862/attachments/1366614/2073007/sipm_condition_analysis.pdf
If there are difference between them (2.8 mm and 3.3 mm) and if the template differ from the old one (GT22 vs GT23) the old average pulse will perform worst here.

abdoulline commented:
would be nice to see "typical" 10TS fC distribution _above_ "phase 
transition" at ~20 GeV with M2 fit overlayed
(that was always a very helpful consideration by Katharina/Stephanie 
previously) to have a guesstimate how it looks like (and which direction to 
modify it) before we go for changing _reco_ pulse shape.
I do think we'd need a(n appropriately) adjusted reco pulse for 2017 HE, 
like we have it for HBHE Run 2 now.

My guess is - we don't have a big difference (pulse-wise) between SiPMs 
types, but Kevin is certainly in a better postion than me to clarify this 
point.
While we obviously have quite quite some difference between shape 203 and 
what M2 fits (with several steps in between simulation and reco) and need 
to deal/cope with this.



On Thu, 17 Nov 2016, mariadalfonso wrote:

> 
> @kpedro88 @abdoulline
> Do the current template 203 need to be updated to follow the analysis of the 2.8 mm and 3.3 mm adjustements
> you have here ?
> https://indico.cern.ch/event/586254/contributions/2361862/attachments/1366614/2073007/sipm_condition_analysi
> s.pdf
> If there are difference between them (2.8 mm and 3.3 mm) and if the template differ from the old one (GT22 vs
> GT23) the old average pulse will perform worst here.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[AEx02sTpihqaNDjFwnc2cOjCcJyoPA2Lks5q_CGZgaJpZM4KxFgT.gif]
> 
> 
>

kpedro88 commented:
The recent SiPM parameter updates have no impact on the pulse shape.

mariadalfonso commented:
@slava77 

Along with your worries for the high chi2, I checked with @sbrand for the Phase2 (high PU and siPM) and the M2 template was always activated
https://github.com/cms-sw/cmssw/blob/CMSSW_6_2_X_SLHC/RecoLocalCalo/HcalRecProducers/python/HcalSimpleReconstructor_hbhe_cfi.py#L31

I will check how this perform with the single particle with the nonPU/PU scenario and let you know

slava77 commented:
is this really needed in 81?
The updates are not really bug fixes.
May I suggest we just continue in 90X (which can/will be used to build 82X at some point soon)

kpedro88 commented:
@slava77 well, the changes for `ts4Max` and `pedSigmaSiPM` are necessary to be consistent with conditions that did go into 81X. In addition, my understanding was that 81X 2017 DIGI-RECO could be used to produce some calibration samples etc. If that is still the case, we need to have the best 2017 HCAL RECO in 81X. (I thought the "virtual" branch 82X would only be used for releases for Phase2 samples.)

slava77 commented:
On 11/17/16 11:47 AM, Kevin Pedro wrote:
> @slava77 <https://github.com/slava77> well, the changes for |ts4Max| and
> |pedSigmaSiPM| are necessary to be consistent with conditions that did
> go into 81X. In addition, my understanding was that 81X 2017 DIGI-RECO
> could be used to produce some calibration samples etc. 

IMO, until M2 chi2 goes below ~100, any serious calibrations are a waste
of time


> If that is still
> the case, we need to have the best 2017 HCAL RECO in 81X. (I thought the
> "virtual" branch 82X would only be used for releases for Phase2 samples.)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261349591>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbh0_mLDaAv4VXSilY82Yo8F1KbWxks5q_K9rgaJpZM4KxFgT>.
>

kpedro88 commented:
That doesn't really address the question of which release will be used for 2017 calibration samples.

slava77 commented:
On 11/17/16 12:49 PM, Kevin Pedro wrote:
> That doesn't really address the question of which release will be used
> for 2017 calibration samples.

no, indeed

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261365267>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbmnOeQUICqdf_K9A74q1kkovb3Uhks5q_L3pgaJpZM4KxFgT>.
>

mariadalfonso committed:
restore chi2=15 for siPM as in the pre_16

cmsbuild commented:
Pull request #16569 was updated. @perrotta, @cmsbuild, @silviodonato, @cvuosalo, @fwyzard, @Martin-Grunewald, @slava77, @davidlange6 can you please check and sign again.

mariadalfonso commented:
@slava77 

my preference goes in merging this PR for the 8_1_0 with the changes with the parameters:
1)  ts4Max                = cms.vdouble(100.,45000.), 
2)  pedSigmaSiPM          = cms.double(0.00065), 
3)  ts4chi2               = cms.vdouble(15.,15.),  

We will basically restore the energy performances before the last minute changes in the conditions GT23 and we should get the same as https://github.com/cms-sw/cmssw/pull/15959
We were also happy the results we got there in terms of dijet scale/met …

This will also make easier to get any feedback from the pfgroup and jetmet since the 20GeV step function in know to everybody since run2 history and this point 20GeV was chosen after many iteration.
If we do not merge this PR we will get at 30GeV, that will only create confusion. 

Maria

slava77 commented:
On 11/17/16 2:02 PM, mariadalfonso wrote:
> @slava77 <https://github.com/slava77>
> 
> my preference goes in merging this PR for the 8_1_0 with the changes
> with the parameters:
> 1) ts4Max = cms.vdouble(100.,45000.),
> 2) pedSigmaSiPM = cms.double(0.00065),
> 3) ts4chi2 = cms.vdouble(15.,15.),

OK, sounds reasonable.


> 
> We will basically restore the energy performances before the last minute
> changes in the conditions GT23 and we should get the same as #15959
> <https://github.com/cms-sw/cmssw/pull/15959>
> We were also happy the results we got there in terms of dijet scale/met …
> 
> This will also make easier to get any feedback from the pfgroup and
> jetmet since the 20GeV step function in know to everybody since run2
> history and this point 20GeV was chosen after many iteration.
> If we do not merge this PR we will get at 30GeV, that will only create
> confusion.
> 
> Maria
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261383924>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbmlyFBsW5bHSEhLt-TufpfvdiHtEks5q_M7sgaJpZM4KxFgT>.
>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

abdoulline commented:
Slava, why do you consider Chi2 downsizing as a pre-requisite, while 
energy scale is very reasonable now?
If Chi2 story will not happily end soon (Maria mentioned it may stay yet 
for some time), while the plan was (yet a week ago) to definitely make the 
samples with 81X, what we (all) do then? 
I agree improving Chi2 (e.g. by improving pulse shape template) might 
change a bit the scale. So we'd need to re-reco these samples. It's kind of 
"deja vu" of end-2014/biginning-2015 situation with M2 
development/finalizing.
But is there a better plan?



On Thu, 17 Nov 2016, Slava Krutelyov wrote:

> On 11/17/16 12:49 PM, Kevin Pedro wrote:
> > That doesn't really address the question of which release will be used
> > for 2017 calibration samples.
> 
> no, indeed
> 
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261365267>, or
> > mute the thread
> ><https://github.com/notifications/unsubscribe-auth/AEdcbmnOeQUICqdf_K9A74q1kkovb3Uhks5q_L3pgaJpZM4KxFg
> T>.
> >
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[AEx02gun7BAmd7CFeKaXBnwzMcYdH3Lxks5q_L43gaJpZM4KxFgT.gif]
> 
> 
>

abdoulline commented:
I still think the only fast enough way to improve Chi2 situation is 
(1) to assess current M2 residuals for HE 2017 (as Kevin mentioned elsewhere). 
I can only second him and (again) to ask Maria (and her MIT colleagues, if available) to do that.
(2) and then to modify M2 template specifically for 2017 HE depending on (1) results.

slava77 commented:
On 11/17/16 2:09 PM, Salavat Abdullin wrote:
> 
> Slava, why do you consider Chi2 downsizing as a pre-requisite, while
> energy scale is very reasonable now?

[Note: I did not mean that getting chi2 of the fit is a prerequisite of
this PR;
I think it will be a prerequisite of "good enough M2 for 2017" as we
converge on the release useful to run calibrations]

because at these large values of chi2 we are just lucky that the scale
looks reasonable.
Some tiny shifts in future MC updates or variations relevant to data can
change this a lot.
huge chi2 values mean  that the input shapes for the fit do not have
much to do with the expected shapes.
Due to the switch in parameters, the worst place is around 20 GeV where
response has apparently ~10% or more discontinuity which depends on PU
and tiny shape variations.

> If Chi2 story will not happily end soon (Maria mentioned it may stay yet
> for some time), while the plan was (yet a week ago) to definitely make the
> samples with 81X, what we (all) do then?

Check the schedule for the next 6-8 months shown in PPD and other meetings.
Central calibration samples at POG level are not in place until 90X is
made in February.

> I agree improving Chi2 (e.g. by improving pulse shape template) might
> change a bit the scale. So we'd need to re-reco these samples. It's kind of
> "deja vu" of end-2014/biginning-2015 situation with M2
> development/finalizing.
> But is there a better plan?
> 
> 
> 
> On Thu, 17 Nov 2016, Slava Krutelyov wrote:
> 
>> On 11/17/16 12:49 PM, Kevin Pedro wrote:
>> > That doesn't really address the question of which release will be used
>> > for 2017 calibration samples.
>> 
>> no, indeed
>> 
>> >
>> > —
>> > You are receiving this because you were mentioned.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261365267>, or
>> > mute the thread
>> ><https://github.com/notifications/unsubscribe-auth/AEdcbmnOeQUICqdf_K9A74q1kkovb3Uhks5q_L3pgaJpZM4KxFg
>> T>.
>> >
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub, or mute the
>> thread.[AEx02gun7BAmd7CFeKaXBnwzMcYdH3Lxks5q_L43gaJpZM4KxFgT.gif]
>> 
>> 
>>
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261385695>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbkUmQvLZ64m5gb0gSZCbhqXPrcRxks5q_NCWgaJpZM4KxFgT>.
>

abdoulline commented:
Well, (in turn) I didn't say you consider Chi2 as a prerequisite for PR, as you clearly said about calibration samples:  
>  until M2 chi2 goes below ~100, any serious calibrations are a waste of time

Good energy scale with bad Chi2 can be a result of both incorrect sigma and imperfect template shape.
For me a good sign is that M2 vs M0 is quite decent. We do need more input from Maria (and her colleagues) in this domain... 

OK, if there is no more plans for 81X samples, it will give us some time for Chi2 "mitigation", we just need to act fast enough.

mariadalfonso commented:
@abdoulline 

The BIG picture is the following: we have 
1) DIGI 2) pulse template 3) a noise estimate.
the chi2 is basically [1-2]*[1-2]/3*3 .
the three things are not perfectly compatible (, where already not perfectly compatible in the run2 neither)

Something might be out of place in the DIGI, not accounted in the noise terms, the pulse doesn't match the DIGI . If we debug these one by one separately and not post M2 machinery will be certainly helpfull.

I'm an offering plots and all my physics intuition since three months, many things were already spotted let's continue.

kpedro88 commented:
@mariadalfonso the digi simulation has already been debugged quite thoroughly (with the exception of TimeSlew as we've discussed). Blindly throwing around suggestions is not going to help us make progress, especially with this tight schedule. If the M2 fit is not performing well, we need to know *exactly* how and why it is not performing. There is no other path forward.

cmsbuild commented:
+1
Tested at: 419ae2ce5441d4272f022e136629a39a09142d55
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16455/summary.html

cmsbuild commented:
Comparison job queued.

mariadalfonso commented:
@kpedro88 
please let's stay constructive

kpedro88 commented:
Continually blaming everything on the digi simulation and refusing to provide long-requested M2 residuals is not constructive.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16569/16455/summary.html

Alternative comparison was/were failed for workflow(s):
140.53

mariadalfonso commented:
@kpedro88 

I'm not blaiming anyone and anything: I never said neither DIGI are crap neither M2 doensn't work.
because I think both are doing a pretty good job.

I'm trying to ask question and I'm trying to build a direction of work so that we can improve things in short term and can be sustainable in long term.

slide 5 
https://indico.cern.ch/event/578882/contributions/2345265/attachments/1358776/2055215/dalfonso_oct20_Phase1.pdf
those are YOUR plots: the residual is just the ratio of those
--> the template match the DIGI in the low energy regime
--> is not good enough at high energy
I'm not refusing to do work, is that you made the plots already 1 month ago

anyway, as I said let's continue to work.

abdoulline commented:
@mariadalfonso 

>The BIG picture is the following: we have 
>1) DIGI 2) pulse template 3) a noise estimate.
> the chi2 is basically [1-2][1-2]/33 

Well, we all know it... My (long time repetitive) suggestions/requests to you (assuming you still can get some help for your MIT colleagues?)  were to do what they Katharina/Stephanie have been doing in 2014/2015:  

(i) to look at 10TS vs fit to assess residuals of M2 
(out of the box, and nowadays trying several "no" options: no TimeSlewSim,  no M2 TimeSlew accounting/recovery and their combination for better understanding of those time-modifying components)   

So that we could then consider/discuss (and apply after some iterations)
(ii)  modifications of reco pulse template [*] to reduce Chi2 
------------------------------------------
[*]
As I've already mentioned it elsewhere:
we did it in 2011 for HBHE reco template yet for M0, using residual estimates of reco pulse (now known as "105" after tuning)  shape wrt DIGI shape from both mc and data (with slightly different phase adjustment for out-of-window correction between them). 
Now we need to use MC DIGI.

abdoulline commented:
@slava77 
may we get it approved, so that we can move on in 90X, hoping to improve the situation with pulse template (and Chi2) in the near future?

slava77 commented:
On 11/18/16 4:44 AM, Salavat Abdullin wrote:
> @slava77 <https://github.com/slava77>
> may we get it approved, so that we can move on in 90X, hoping to improve
> the situation with pulse template (and Chi2) in the near future?

I still need to check the results


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261523704>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbiCgjKbAHIBIrSshwFz-Nkbez_Ltks5q_Z3IgaJpZM4KxFgT>.
>

slava77 commented:
on the topic of chi2: how much contribution do we get from TS0,1,2,7,8,9? something that we don't really fit to more than just a pedestal in the model.
This is more specific for PU case.

mariadalfonso commented:
@slava77 

I updated slide9/10 with the average DIGI profile.
Splitted in the highEnergy and lowEnergy and also compared the PU and noPU (separately for HE and HB) http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf

It seems that there is a lot of energy on average outside the TS45 for the low Energy rechit even in the nonPU scenario. I will dig more.

slava77 commented:
On 11/18/16 11:35 AM, mariadalfonso wrote:
> @slava77 <https://github.com/slava77>
> 
> I updated slide9/10 with the average DIGI profile.
> Splitted in the highEnergy and lowEnergy and also compared the PU and
> noPU (separately for HE and HB)
> http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
> 
> It seems that there is a lot of energy on average outside the TS45 for
> the low Energy rechit even in the nonPU scenario. I will dig more.

5 GeV is a lot of energy for HB
not so much for the forward HE.
So HB vs HE comparison might be pretty skewed.

Probably a more meaningful comparison would be if the threshold was
scaled in pt
... another option is to compare with what we think worked: look at 2016
HE shapes.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261621669>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbnGe4lN5luVt2WHx_6t6_YTGJP39ks5q_f4MgaJpZM4KxFgT>.
>

slava77 commented:
On 11/18/16 12:07 PM, Slava Krutelyov wrote:
> On 11/18/16 11:35 AM, mariadalfonso wrote:
>> @slava77 <https://github.com/slava77>
>>
>> I updated slide9/10 with the average DIGI profile.
>> Splitted in the highEnergy and lowEnergy and also compared the PU and
>> noPU (separately for HE and HB)
>> http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf
>>
>> It seems that there is a lot of energy on average outside the TS45 for
>> the low Energy rechit even in the nonPU scenario. I will dig more.
> 
> 5 GeV is a lot of energy for HB
> not so much for the forward HE.
> So HB vs HE comparison might be pretty skewed.
> 
> Probably a more meaningful comparison would be if the threshold was
> scaled in pt
> ... another option is to compare with what we think worked: look at 2016
> HE shapes.

I was thinking about the case with PU where real energy can show up
outside of TS45 from OOT
when I made the comment.

p10 the rhs blue-red looks somewhat flat outside of TS45 and up to TS7.
TS8,9 should be depleted in MC because there is no OOT PU there (we mix
up to +3).

I don't quite understand the slope on the low end without PU either. (p9)


> 
> 
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261621669>, or
>> mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AEdcbnGe4lN5luVt2WHx_6t6_YTGJP39ks5q_f4MgaJpZM4KxFgT>.
>>
>

abdoulline commented:
@slava77  - concerning your fair point about 10TS:  some time ago Igor Volobouev has rised this question and we all agreed on a need to _consider_ (study) reducing the range to possibly 4 TS centered at trig.TS (TS#4 in case of HE)      

@mariadalfonso - please, make sure (going to DAS and checking prod.configs) that what labeled as 0PU sample is really no-PU. As far as I remember, regular 25ns RelVals had (so far) PU35,  now there are possibly both PU35 and new PU50? - Cannot check is from home (no certificate to access DAS...)

kpedro88 commented:
I agree with @abdoulline, all of those HE pulse plots look like what I expect from PU. Remember, with the new depth segmentation, the first few depths are quite sensitive to PU deposits. The same behavior can be observed with HPDs (using hardcode conditions to set HE QIEType to 0 and generate average pedestals, gains, etc. appropriate for HPDs for all the new channels).

slava77 commented:
On 11/18/16 12:22 PM, Salavat Abdullin wrote:
> @slava77 <https://github.com/slava77> - concerning your fair point about
> 10TS: some time ago Igor Volobouev has rised this question and we all
> agreed on a need to /consider/ (study) reducing the range to possibly 4
> TS centered at trig.TS (TS#4 in case of HE)

OK. {I was mostly trying to understand what adds to the chi2: center or
tails; not aiming to just cut it down as is}
3-pulse fit and even 2-pulse is not really a fit with just 4 points,
unless time is fixed and pedestal comes from outside TS.

> 
> @mariadalfonso <https://github.com/mariadalfonso> - please, make sure
> (going to DAS and checking prod.configs) that what labeled as 0PU sample
> is really no-PU. As far as I remember, regular 25ns RelVals had (so far)
> PU35, now there are possibly both PU35 and new PU50? - Cannot check is
> from home (no certificate to access DAS...)

dataset name "PU25ns_81X_upgrade2017_realistic_v22-v1" indeed appears to
correspond to the default PU35

/RelValTTbar_13/CMSSW_8_1_0_pre16-81X_upgrade2017_realistic_v22-v1/GEN-SIM-DIGI-RAW
is the no-PU sample



> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261632451>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbtGDZfhsTUQbQ8xyQRsTbqp05TGOks5q_gj_gaJpZM4KxFgT>.
>

abdoulline commented:
@slava77 - we definitely need to find a balance between not destroying 3-pulse fit and minimize rather high (especially in innermost HE rings) PU contribution to Chi2. 4TS might be suboptimal, indeed.

slava77 commented:
Pi gun plot:
baseline
![orig_he_eraw_vs_e_max50](https://cloud.githubusercontent.com/assets/4676718/20482950/6d433538-afa4-11e6-81d2-76e48f99fd0e.png)

with this PR
![sign797_he_eraw_vs_e_max50](https://cloud.githubusercontent.com/assets/4676718/20482939/5b686ad6-afa4-11e6-8cb9-41d8ce4fab27.png)

and almost no effect above the switch-over point.

This confirms expected behavior and shows the needed change.

CPU in M2 increased a bit ``` 3240.43 ms/ev ->      3382.07 ms/ev hbheprereco ```

Higher level quantities are not significantly affected by the change: the largest effect on jet response is in samples with PU
ttbar+PU35
<img width="857" alt="wf10224_ak4pfchs_cogen_e" src="https://cloud.githubusercontent.com/assets/4676718/20485999/de7f5084-afb2-11e6-8901-ab8fa16b0d43.png">

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/16569 419ae2c
- M2 adjusted to changes in response of SiPMs
- jenkins tests pass and comparisons with baseline show changes in 2017 and 2023 workflows as expected
- local tests confirm expected behavior (see  https://github.com/cms-sw/cmssw/pull/16569#issuecomment-261950920 and http://dalfonso.web.cern.ch/dalfonso/dalfonso_siPM_M2_parameters.pdf )

abdoulline commented:
@davidlange    David, may we have it in 810? 
I hope our HLT colleagues  @fwyzard, @Martin-Grunewald
have no objection about 2017 HE improvements

hatakeyamak commented:
Thanks Slava. Your plots look consistent with the HCAL internal check with 0PU pion gun, but now your confirm good behavior in events with PU.

810pre16+PR16544+ PR16569 - 2017 w.r.t. 810pre16 default - 2016
[2] http://hep.baylor.edu/cmshcal/kmandal/81X_2017_PR16544_16569_vs_810pre16_2016_SinglePi/

810pre16+PR16544 - 2017 w.r.t. 810pre16 default - 2016
[1] http://hep.baylor.edu/cmshcal/kmandal/81X_2017_PR16544_vs_810pre16_2016_SinglePi/

810pre16 default - 2017 w.r.t. 810pre16 default - 2016
[0] http://hep.baylor.edu/cmshcal/kmandal/810pre16_2017_vs_810pre16_2016_SinglePi/

Martin-Grunewald commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

davidlange6 commented:
+1
